### PR TITLE
Add recipe for unitig-counter

### PR DESCRIPTION
**Pull request by johnlees**
:information_source:
Bioconda has finished the [GCC7 migration](https://github.com/bioconda/bioconda-recipes/issues/13578). If you are dealing with C/C++ or Python package it can be that you need to rebuild other dependent packages. [Bioconda utils - update-pinning](https://bioconda.github.io/updating.html#updating-recipes-for-a-pinning-change) will assist you with that. If you have any questions please use [issue 13578](https://github.com/bioconda/bioconda-recipes/issues/13578).

----------------

* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [x] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).

### CODE REVIEW

**Comment by johnlees** [comment]
Could I have a review please, @bioconda/core 

[This adds a program to count unitigs in bacterial populations, based off the DBGWAS software.]

I had to turn off OS X as the Clang is v4.0, and >4.5 is required to compile. I think it would work with GCC on OS X though. Is that possible, or is skipping OS X the best thing to do in this case?

File: recipes/unitig-counter/build.sh [hunk 1]
```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+# set up directory for building boost compiled dependencies
+# see
+# https://github.com/conda-forge/boost-cpp-feedstock/blob/master/recipe/build.sh
+# https://github.com/boostorg/program_options/blob/develop/.travis.yml
+git clone --depth 1 --single-branch --branch boost-1.69.0 https://github.com/boostorg/boost.git boost
```
Comment (dpryan79) [inline recipes/unitig-counter/build.sh]: Can't you use boost from conda-forge?

File: recipes/unitig-counter/build.sh [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    cxxflags="${CXXFLAGS}" \
+    linkflags="${LINKFLAGS}" \
+    --layout=system \
+    -j"${CPU_COUNT}" \
+    install
+popd
+
+# get submodules
+rmdir gatb-core
+git clone https://github.com/GATB/gatb-core.git
```
Comment (dpryan79) [inline recipes/unitig-counter/build.sh]: It's often easier to use a `git_url` and version so submodules are handled for you. You'll have to add a skip_lint for that, but it's easier.

File: recipes/unitig-counter/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+build:
+  number: 0
+  skip: True #  [osx]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - python <3
```
Comment (dpryan79) [inline recipes/unitig-counter/meta.yaml]: Please put python in the host section.

File: recipes/unitig-counter/build.sh (hunk 1 above)
Comment (johnlees) [reply recipes/unitig-counter/build.sh]: Yes, this does work with this one! Changed to do that

File: recipes/unitig-counter/meta.yaml (hunk 3 above)
Comment (johnlees) [reply recipes/unitig-counter/meta.yaml]: Not needed, now boost is being pulled from conda-forge

File: recipes/unitig-counter/build.sh (hunk 2 above)
Comment (johnlees) [reply recipes/unitig-counter/build.sh]: Done

**Comment by johnlees** [comment]
@dpryan79 This build timed out, and so didn't give any diagnostics in the log. This didn't happen for me when I ran the new commit build locally. Any suggestions? Thanks!

**Comment by dpryan79** [comment]
@johnlees Did you run it locally in docker?

**Comment by johnlees** [comment]
No, just with conda build

**Comment by johnlees** [comment]
Also odd because very little changed between this and the previously working commit.

**Comment by dpryan79** [comment]
Try testing with `bioconda-utils build --docker` instead, it's what the real build is using.

**Comment by johnlees** [comment]
Problem was ssh git submodules, leading to hanging waiting for a password. Changed to https now

**Pull request merged**